packages/labs/gen-wrapper-react/package.json

```diff
@@ -60,11 +60,13 @@
     }
   },
   "dependencies": {
+    "@lit-internal/tests": "0.0.0",
     "@lit-labs/analyzer": "^0.2.0",
     "@lit-labs/gen-utils": "^0.1.0"
   },
   "devDependencies": {
-    "@lit-internal/tests": "^0.0.0"
+    "@lit-internal/tests": "^0.0.0",
+    "@types/react": "^18.0.17"
```

Comment:
I understand that this fixed the type compilation issue, but I don't understand why. The `gen-wrapper-react` source does not actually do anything with react (note, it doesn't have react as a dependency), it only generates and tests code that uses react.

The generated react wrapper has a peer dependency on `@types/react` (see [generator](https://github.com/lit/lit/blob/main/packages/labs/gen-wrapper-react/src/index.ts#L97) and [golden](https://github.com/lit/lit/blob/main/packages/labs/gen-wrapper-react/goldens/test-element-a/package.json#L15)), and the test folder into which the generated wrapper is installed depends on `@types/react` [[source](https://github.com/lit/lit/blob/main/packages/labs/gen-wrapper-react/test-output/package.json#L10)].

So it's unclear in my mind what is resolving `@types/react` down into `gen-wrapper-react`'s node_modules.  I want to understand that, otherwise it's not clear we're missing something that will hit actual users, since the direct user of the generate wrapper having a dep on `@types/react` apparently isn't sufficient.

Reply:
Oh nice, i think you just unlocked something for me!

in the lit-labs/react pacakge.json there's a `types` property
```
{
...
    "types": ["react", "react-dom", "mocha"]
}
```

and when i add that to the `lit-labs/gen-wrapper-react/test-output/package.json`
```
{
...
    "types": ["react", "react-dom", "mocha"]
}
```

We get passing tests for `gen-wrapper-react` without adding `@types/react` to the gen-wrapper-react package.

Reply:
That confuses me. The `"types"` field is supposed to limit what `@types/` packages are included, not expand them: https://www.typescriptlang.org/tsconfig#types

Reply:
even more confusing it fails windows tests but passes others

Reply:
without the `types` array, all local tests fail on unix and windows

Reply:
I'm trying to discern whether we are operating within the domain of our npm workspace or if the test-output is completely separate from the workspace.

Adding the @types resolves this but AFAIU @kevinpschaaf we don't want the @types associated with `gen-wrapper-react` only the `test-output`.

This feels in the domain of ownership and npm workspaces. It makes me think that `gen-wrapper-react` is the domain and `test-output` is operating within that domain.

The `test-output` is independent and sort of an integration test for the wrapper no? But do we really know how isolated `test-output` from `gen-wrapper-react`? That's currently where I'm at trying to rationalize why adding `@types` to `gen-wrapper-react` works here.

It would make me wonder if `test-output` was entirely outside the Lit package (a real world use case) would resolve this.

Reply:
is it possible that we need to make `test-output` a workspace?

Reply:
resolved with https://github.com/lit/lit/pull/3310